src/main/java/sirius/db/mongo/Finder.java

```diff
@@ -371,16 +375,43 @@ public long countIn(Class<?> type) {
      * Counts the number of documents in the result of the given query.
      * <p>
      * Note that limits are ignored for this query.
+     * If the there are no filters in this query, an estimate is returned instead.
      *
      * @param collection the collection to search in
      * @return the number of documents found
      */
     public long countIn(String collection) {
+        return countIn(collection, false, 0).orElse(0L);
+    }
+
+    /**
+     * Counts the number of documents in the result of the given query.
+     * <p>
+     * Note that limits are ignored for this query.
+     * If there are no filters in this query and forceAccurate is false, a pre-counted estimate is returned instead.
+     *
+     * @param collection    the collection to search in
+     * @param forceAccurate if set to <tt>true</tt> we'll never use <b>estimatedDocumentCount</b> which is way more efficient but might return wrong values in case a cluster is active which had experienced an unclean shutdown.
+     * @param maxTimeMS     the maximum process time for this cursor in milliseconds, 0 for unlimited
+     * @return the number of documents found, wrapped in an Optional, or an empty Optional if the query timed out
+     */
+    public Optional<Long> countIn(String collection, boolean forceAccurate, long maxTimeMS) {
         Watch w = Watch.start();
         try {
-            return mongo.db()
-                        .getCollection(collection)
-                        .countDocuments(filterObject, new CountOptions().collation(mongo.determineCollation()));
+            if (filterObject.isEmpty() && !forceAccurate) {
+                return Optional.of(mongo.db()
+                                        .getCollection(collection)
+                                        .estimatedDocumentCount(new EstimatedDocumentCountOptions().maxTime(maxTimeMS,
+                                                                                                            TimeUnit.MILLISECONDS)));
```

Comment:
the max time for the estimate is not really useful, as even with a huge collection this is always super fast and there is probably not even an interrupt point in the operation, as even with 1ms this never hit the time out - but i think it's still a good idea to include as the option is available